From ceb3d65e84c39a3bee02662e4284efb87945e534 Mon Sep 17 00:00:00 2001
From: Tanmay Jagdale <tanmayj@broadcom.com>
Date: Wed, 10 Dec 2014 11:56:52 +0530
Subject: [PATCH 0609/1532] xlp4xx: fix the xlp4xx-linux-kernel.dts file

Added the following fixes to xlp4xx-linux-kernel.dts
  - Added the chosen section
  - Removed the memory node
  - Removed the spi node
  - Removed the gpio node
[Based on SDK 3.2]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 arch/mips/boot/dts/xlp4xx-linux-kernel.dts | 83 +++---------------------------
 1 file changed, 6 insertions(+), 77 deletions(-)

diff --git a/arch/mips/boot/dts/xlp4xx-linux-kernel.dts b/arch/mips/boot/dts/xlp4xx-linux-kernel.dts
index d3900c2..8363451 100644
--- a/arch/mips/boot/dts/xlp4xx-linux-kernel.dts
+++ b/arch/mips/boot/dts/xlp4xx-linux-kernel.dts
@@ -9,13 +9,6 @@
 	#address-cells = <2>;
 	#size-cells = <2>;
 
-	memory {
-		device_type = "memory";
-		reg =  <0 0x00100000 0 0x0FF00000	// 255M at 1M
-			0 0x20000000 0 0xa0000000	// 2560M at 512M
-			0 0xe0000000 1 0x00000000>;
-	};
-
 	soc {
 		#address-cells = <2>;
 		#size-cells = <1>;
@@ -136,76 +129,12 @@
 				};
 			};
 		};
+	};
 
-		gpio: xlp_gpio@34000 {
-			compatible = "netlogic,xlp-gpio";
-			reg = <0 0x34100 0x1000>;
-			#gpio-cells = <2>;
-			gpio-controller;
-
-			#interrupt-cells = <1>;
-			interrupt-parent = <&pic>;
-			interrupts = <39>;
-			interrupt-controller;
-		};
-
-		spi: xlp_spi@3a000 {
-			compatible = "netlogic,xlp-spi";
-			#address-cells = <1>;
-			#size-cells = <0>;
-			reg = <0 0x3a000 0x100>;
-
-			interrupts = <34>;
-			interrupt-parent = <&pic>;
-
-			spi_nor@1 {
-				compatible = "spansion,s25sl12801";
-				#address-cells = <1>;
-				#size-cells = <1>;
-
-				reg = <1>;		/* Chip Select */
-				spi-max-frequency = <40000000>;
-
-				partition@0 {
-					label = "x-loader";
-					reg = <0x0 0x200000>; /* 2M */
-					read-only;
-				};
-
-				partition@200000 {
-					label = "kernel";
-					reg = <0x200000 0x500000>; /* 5M */
-				};
-
-				partition@700000 {
-					label = "rootfs";
-					reg = <0x700000 0x800000>; /* 8M */
-				};
-
-				partition@f00000 {
-					label = "env";
-					reg = <0xf00000 0x100000>; /* 1M */
-					read-only;
-				};
-			};
-
-			spi_nand@2 {
-				compatible = "spinand,mt29f";
-				#address-cells = <1>;
-				#size-cells = <1>;
-				reg = <2>;	/* SPI NAND CS */
-				spi-max-frequency = <50000000>;
-
-				partition@0 {
-					label = "firmware";
-					reg = <0x0 0x1c00000>;
-				};
-
-				partition@1 {
-					label = "kernel";
-					reg = <0x1c00000 0x0>;
-				};
-			};
-		};
+	chosen {
+		bootargs = "console=ttyS0,115200 rhash_entries=1048576";
+		netlogic,cpumask = <0xffffffff>;
+		netlogic,reserve-mem = <256>;
+		netlogic,spill-mem = <32>;
 	};
 };
-- 
1.9.1

